virtcontainers/qemu.go

```diff
@@ -653,9 +654,24 @@ func (q *qemu) stopSandbox() error {
 		return err
 	}
 
-	err = os.RemoveAll(filepath.Join(RunVMStoragePath, q.id))
+	// cleanup vm path
+	dir := filepath.Join(RunVMStoragePath, q.id)
+
+	// If it's a symlink, remove both dir and the target.
+	// This can happen when vm template links a sandbox to a vm.
+	link, err := filepath.EvalSymlinks(dir)
 	if err != nil {
-		q.Logger().WithError(err).Error("Fail to clean up vm directory")
+		q.Logger().WithError(err).Warnf("failed to resolve vm path %s", dir)
+	}
+	q.Logger().WithField("link", link).WithField("dir", dir).Infof("cleanup vm path")
+
+	if err = os.RemoveAll(dir); err != nil {
```

Comment:
nit, niggle - the `; err != nil` idiom here is not used above - maybe as we only have the one return val?, but it might be nice to keep the flows the same?